search-books/src/main/java/nix/education/service/response/SolrResponseService.java

```diff
@@ -0,0 +1,115 @@
+package nix.education.service.response;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.SneakyThrows;
+import nix.education.repository.BookRepository;
+import nix.education.service.BookDataService;
+import nix.education.service.query.solr.SolrQueryBuilderService;
+import nix.education.util.SearchConstants;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.response.FacetField;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.RangeFacet;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SolrResponseService implements ResponseService {
+
+    @Autowired
+    @Qualifier("solrQueryBuilderServiceImpl")
```

Comment:
Do we have more then one impl? Why to specify the qualifier?

Reply:
Yes. agree. I just forgot delete this annotation.